### DISCUSSION TITLE
PHP 8.1 Requirement in the Future

### DISCUSSION THREAD

**Discussion by jbtronics** [Part-DB/Part-DB-server#280] [Announcements]
In the next months (maybe around June/July), I will migrate Part-DBs codebase to use PHP 8.1 (from currently 7.4).

More and more libraries, Part-DB is using, require at least PHP 8.1 and the versions of them Part-DB has to use (to still support PHP 7.4) are quite old and unmaintained. This is a potential security risk (even if I am not aware of any possible exploits), and makes it harder to maintain Part-DB (as sometimes some workaround are needed to make the code work).
Besides that the newer PHP versions offers some new features (especially union types, enums and native attributes), which could highly improve Part-DBs code and make it easier to maintain.

Part-DB will then require a PHP version of PHP 8.1, which means that it will not work on servers where only PHP 7.4 or PHP 8.0 is available. All current versions of the major linux distributions should ship PHP 8.1 or higher then (Debian 12 should be released around that time in summer 2023), and older ones can use PHP 8.1 via backport repositories.
The survey results say that the big majority of Part-DB users are already using PHP8.1 or higher; either by native installation or by using the docker image (which will always ship at least the required PHP version). So this will hopefully not impact too many Part-DB users.

**Comment by Pyromane** [comment]
Thank you for the advance notice!
